src/proxy/routes-engine.ts

```diff
@@ -3,34 +3,62 @@ import http from "http";
 import { DEFAULT_CONFIG } from "../config";
 import { decodeCookie, findFile } from "../core/utils";
 
-export const handleUserCustomRoutes = async (folder: string): Promise<UserDefinedRoute[]> => {
+export const handleUserConfig = async (folder: string): Promise<UserConfig> => {
```

Comment:
Refactored UserDefinedRoutes to UserConfig to handle all nodes inside the new SWA configuration file.